### PR TITLE
LocalTransaction: Try to address the flakeyness of the test

### PR DESCRIPTION
**Pull request by Geod24**
Looking at the test, it seems to be the same cause as @hewison-chris has been fixing all over the code base, so giving a try to the same solution.

### CODE REVIEW

File: source/agora/test/LocalTransactions.d [hunk 1]
```diff
@@ -73,9 +73,10 @@ unittest
     // Create 6 transactions - one for each validator
     auto txs = blocks[0].spendable().map!(txb => txb.sign());
     // Distribute them to different clients
-    txs.takeExactly(6).enumerate.each!((idx, tx) =>
-        network.clients[idx % network.clients.length]
-            .putTransaction(tx));
+    txs.takeExactly(6).enumerate.each!(
+        (idx, tx) {
+            network.ensureTxInPool(tx, (idx % network.clients.length).only);
```
Comment (omerfirmak) [inline source/agora/test/LocalTransactions.d:78]: This defeats the purpose of the test tho, no?
We want nodes to ask for the missing TXs.

**Review by omerfirmak — CHANGES_REQUESTED** [review]

File: source/agora/test/LocalTransactions.d (hunk 1 above)
Comment (Geod24) [reply source/agora/test/LocalTransactions.d:78]: It only ensures the tx is in the pool of that specific client

Comment (hewison-chris) [reply source/agora/test/LocalTransactions.d:78]: I think @omerfirmak has a point as it is also putting the tx to each.

Comment (omerfirmak) [reply source/agora/test/LocalTransactions.d:78]: But [this ](https://github.com/bosagora/agora/pull/2367/files#diff-2cb6a9d42ca703fefdc67b1c1062e63c52c36cb15156b0e48d4cf36b9ace51e8R1195-R1199) also does a `putTransaction?`

Comment (hewison-chris) [reply source/agora/test/LocalTransactions.d:78]: Or maybe not.

Comment (Geod24) [reply source/agora/test/LocalTransactions.d:78]: @omerfirmak : But only on one index

File: source/agora/test/LocalTransactions.d [hunk 2]
```diff
@@ -86,9 +73,10 @@ unittest
...
```
Comment (hewison-chris) [inline source/agora/test/LocalTransactions.d:78]:
```suggestion
            network.ensureTxInPool(tx.hashFull(), (idx % network.clients.length).only);
```

Comment (hewison-chris) [reply source/agora/test/LocalTransactions.d:78]: ^ But we need the deleted put of tx to each..

**Review by hewison-chris — CHANGES_REQUESTED** [review]

**Comment by hewison-chris** [comment]
@Geod24 can I update this one. I think I need the fix now in a different PR. I'll cherry-pick the commits and see if it solves it.

**Comment by Geod24** [comment]
Sure

**Comment by hewison-chris** [comment]
Tested locally and works well. I'll merge it.

**Review by hewison-chris — APPROVED** [review]
LGTM.

File: source/agora/test/LocalTransactions.d (hunk 1 above)
Comment (hewison-chris) [reply source/agora/test/LocalTransactions.d:78]: It is correct on closer inspection.

**Review by hewison-chris — APPROVED** [review]

**Comment by Geod24** [comment]
Superseded by https://github.com/bosagora/agora/pull/2397

**Pull request closed without merging**